base/compiler/ssair/inlining.jl

```diff
@@ -574,11 +574,11 @@ function batch_inline!(todo::Vector{Any}, ir::IRCode, linetable::Vector{LineInfo
     return ir
 end
 
-function spec_lambda(@nospecialize(atype), sv::OptimizationState, @nospecialize(invoke_data))
+function spec_lambda(interp::AbstractInterpreter, @nospecialize(atype), sv::OptimizationState, @nospecialize(invoke_data))
```

Comment:
get the world out of the optimization state? This function shouldn't take an interpreter because it's in the optimizer.

Reply:
I swear there was some other thing I needed other than the world counter here, but apparently not.  Fixed.